src/include/reconcile_inline.h

```diff
@@ -195,11 +195,11 @@ __wt_rec_need_split(WT_RECONCILE *r, size_t len)
      * worth splitting. We also want to temper this effect to avoid in-memory updates from
      * dominating the calculation and causing excessive splitting. Therefore, we'll limit the impact
      * to a tenth of the cache usage occupied by those updates. For small updates the overhead of
-     * the update structure could skew the calculation, so we subtract the overhead before
-     * considering the cache usage by the updates.
+     * the update structure could skew the calculation. Therefore, we only consider the size of the
+     * value in the saved updates' memory size
      */
     if (r->page->type == WT_PAGE_ROW_LEAF && page_items > WT_REC_SPLIT_MIN_ITEMS_USE_MEM)
-        len += (r->supd_memsize - ((size_t)r->supd_next * WT_UPDATE_SIZE)) / 10;
+        len += r->supd_restore_memsize / 10;
```

Comment:
Should we skip this optimization for in memory database? I feel there is no benefit to split for in memory case as it only leads us to use more memory.